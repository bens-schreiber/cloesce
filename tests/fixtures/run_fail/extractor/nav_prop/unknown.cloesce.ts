// @ts-nocheck

@D1
export class Foo {
  @PrimaryKey
  id: number;

  @DataSource
  foo: IncludeTree<Foo> = { bar: {} }; // bar does not exist
}
