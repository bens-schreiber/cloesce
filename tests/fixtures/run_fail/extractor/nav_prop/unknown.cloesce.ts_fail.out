==== CLOESCE ERROR ====
Error [UnknownNavigationPropertyReference]: Unknown Navigation Property Reference
Phase: TypeScript IDL Extraction
Context: Foo.foo bar
@DataSource
  foo: IncludeTree<Foo> = { bar: {} };

Suggested fix: Verify that the navigation property reference model exists, or create a model.
